sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -241,16 +247,19 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
       spark: SparkSession,
       isMatchedPlan: LogicalPlan => Boolean,
       cascade: Boolean,
-      blocking: Boolean): Unit = {
+      blocking: Boolean): Boolean = {
     val shouldRemove: LogicalPlan => Boolean =
       if (cascade) {
         _.exists(isMatchedPlan)
       } else {
         isMatchedPlan
       }
-    val plansToUncache = cachedData.filter(cd => shouldRemove(cd.plan))
+    var plansToUncache: IndexedSeq[CachedData] = null
```

Comment:
@vrozov why do we need the code change here if this PR is to improve the logging?